daemon/cmd/daemon_main.go

```diff
@@ -584,6 +584,9 @@ func initializeFlags() {
 	flags.Bool(option.EnableSVCSourceRangeCheck, true, "Enable check of service source ranges (currently, only for LoadBalancer)")
 	option.BindEnv(option.EnableSVCSourceRangeCheck)
 
+	flags.Int(option.AddressScopeMax, defaults.AddressScopeMax, "Maximum local address scope for ipcache to consider host addresses")
+	option.BindEnv(option.AddressScopeMax)
```

Comment:
I'd be tempted to just mark it hidden. I don't think that this is something where we've analysed the overall problems clearly enough to expose a model to the user that they can reason about. We can probably even solve the underlying problem just by refactoring the logic and getting it right, without user input.

I see this as a super advanced setting that some individuals may need if they deeply understand the way Cilium works. As a practical solution to an immediate problem, it works, and it provides flexibility to users. But it also feels to me like a shortcut that allows us to avoid addressing tech debt in the immediate term.